tests/components/mqtt_statestream/test_init.py

```diff
@@ -3,375 +3,354 @@
+    """Setup should fail if no base_topic is set."""
+    assert await add_statestream(hass) is False
+
+
+async def test_setup_succeeds_without_attributes(hass, mqtt_mock):
+    """Test the success of the setup with a valid base_topic."""
+    assert await add_statestream(hass, base_topic="pub")
+
+
+async def test_setup_succeeds_with_attributes(hass, mqtt_mock):
+    """Test setup with a valid base_topic and publish_attributes."""
+    assert await add_statestream(hass, base_topic="pub", publish_attributes=True)
+
+
+async def test_state_changed_event_sends_message(hass, mqtt_mock):
+    """Test the sending of a new message if event changed."""
+    e_id = "fake.entity"
+    base_topic = "pub"
+
+    # Add the statestream component for publishing state updates
+    assert await add_statestream(hass, base_topic=base_topic)
+    await hass.async_block_till_done()
+
+    # Reset the mock because it will have already gotten calls for the
+    # mqtt_statestream state change on initialization, etc.
+    mqtt_mock.async_publish.reset_mock()
+
+    # Set a state of an entity
+    mock_state_change_event(hass, State(e_id, "on"))
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    # Make sure 'on' was published to pub/fake/entity/state
+    mqtt_mock.async_publish.assert_called_with("pub/fake/entity/state", "on", 1, True)
+    assert mqtt_mock.async_publish.called
+
+
+async def test_state_changed_event_sends_message_and_timestamp(hass, mqtt_mock):
+    """Test the sending of a message and timestamps if event changed."""
+    e_id = "another.entity"
+    base_topic = "pub"
+
+    # Add the statestream component for publishing state updates
+    assert await add_statestream(
+        hass, base_topic=base_topic, publish_attributes=None, publish_timestamps=True
+    )
+    await hass.async_block_till_done()
+
+    # Reset the mock because it will have already gotten calls for the
+    # mqtt_statestream state change on initialization, etc.
+    mqtt_mock.async_publish.reset_mock()
+
+    # Set a state of an entity
+    mock_state_change_event(hass, State(e_id, "on"))
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    # Make sure 'on' was published to pub/fake/entity/state
+    calls = [
+        call.async_publish("pub/another/entity/state", "on", 1, True),
+        call.async_publish("pub/another/entity/last_changed", ANY, 1, True),
+        call.async_publish("pub/another/entity/last_updated", ANY, 1, True),
+    ]
+
+    mqtt_mock.async_publish.assert_has_calls(calls, any_order=True)
+    assert mqtt_mock.async_publish.called
+
+
+async def test_state_changed_attr_sends_message(hass, mqtt_mock):
+    """Test the sending of a new message if attribute changed."""
+    e_id = "fake.entity"
+    base_topic = "pub"
+
+    # Add the statestream component for publishing state updates
+    assert await add_statestream(hass, base_topic=base_topic, publish_attributes=True)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)